packages/mds-policy/api.ts

```diff
@@ -64,84 +65,105 @@ function api(app: express.Express): express.Express {
+      // TODO filter by start/end applicability
+      const { start_date = now(), end_date = now() } = req.query
+      const { scopes } = res.locals
+
+      try {
+        /*
+          If the client is scoped to read unpublished policies,
+          they are permitted to query for both published and unpublished policies.
+          Otherwise, they can only read published.
+        */
+        const { get_published = null, get_unpublished = null } = scopes.includes('policies:read')
+          ? parseRequest(req, { parser: x => (x ? JSON.parse(x) : null) }).query('get_published', 'get_unpublished')
+          : { get_published: true }
+
+        if (start_date > end_date) {
+          throw new BadParamsError(`start_date ${start_date} > end_date ${end_date}`)
+        }
+        const policies = await db.readPolicies({ get_published, get_unpublished })
+        const prev_policies: UUID[] = policies.reduce((prev_policies_acc: UUID[], policy: Policy) => {
+          if (policy.prev_policies) {
+            prev_policies_acc.push(...policy.prev_policies)
+          }
+          return prev_policies_acc
+        }, [])
+        const active = policies.filter(p => {
+          // overlapping segment logic
+          const p_start_date = p.start_date
+          const p_end_date = p.end_date || Number.MAX_SAFE_INTEGER
+          return end_date >= p_start_date && p_end_date >= start_date && !prev_policies.includes(p.policy_id)
+        })
+
+        if (active.length === 0) {
+          throw new NotFoundError('No policies found!')
+        }
 
-    try {
-      const policies = await db.readPolicies({ get_published, get_unpublished })
-      const prev_policies: UUID[] = policies.reduce((prev_policies_acc: UUID[], policy: Policy) => {
-        if (policy.prev_policies) {
-          prev_policies_acc.push(...policy.prev_policies)
+        res.status(200).send({ version: res.locals.version, policies: active })
+      } catch (error) {
+        if (error instanceof NotFoundError) {
+          return res.status(404).send({ error })
         }
-        return prev_policies_acc
-      }, [])
-      const active = policies.filter(p => {
-        // overlapping segment logic
-        const p_start_date = p.start_date
-        const p_end_date = p.end_date || Number.MAX_SAFE_INTEGER
-        return end_date >= p_start_date && p_end_date >= start_date && !prev_policies.includes(p.policy_id)
-      })
-      res.status(200).send({ version: res.locals.version, policies: active })
-    } catch (err) {
-      res.status(404).send({
-        error: 'not found'
-      })
+        if (error instanceof BadParamsError) {
+          return res.status(400).send({ error })
+        }
+        next(new ServerError(error))
+      }
     }
-  })
+  )
 
-  app.get(pathsFor('/policies/:policy_id'), async (req: PolicyApiRequest, res: GetPolicyResponse) => {
-    const { policy_id } = req.params
-    const { scopes } = res.locals
+  app.get(
```

Comment:
Still needs to be able to support pagination, according to the spec we want to submit.

Reply:
Implementing pagination (properly) is not a small ask, I don't think we can get it done today.